### PR TITLE
Cmake updates

### PR DESCRIPTION
**Pull request by mathstuf**
See https://github.com/compiler-explorer/compiler-explorer/pull/6452

### CODE REVIEW

File: bin/yaml/tools.yaml [hunk 1]
```diff
@@ -9,13 +9,28 @@ tools:
     check_exe: bin/cmake --help
     targets:
       - 3.26.1
-      - 3.26.4
-      - 3.27.9
+      - name: 3.26.4
```
Comment (mattgodbolt) [inline bin/yaml/tools.yaml:12]: Please stick to the format of the others. The `symlink` is actively harmful, and is only used as noted below (to distinguish the one `cmake` we use on the site itself to compile things)

File: bin/yaml/tools.yaml [hunk 2]
```diff
@@ -27,6 +42,7 @@ tools:
     symlink: ninja
     targets:
       - 1.11.1
+      - 1.12.0
```
Comment (mattgodbolt) [inline bin/yaml/tools.yaml:45]: Why the added `ninja` here? Where is it referenced?

File: bin/yaml/tools.yaml [hunk 3]
```diff
@@ -9,13 +9,28 @@ tools:
...
       - 3.28.0-rc5
       - name: 3.28.0
         symlink: cmake  # The one distinguished cmake we use for "cmake" on the site
-      - 3.28.4
-      - 3.29.2
+      - name: 3.28.4
+        symlink: cmake-v3.28
+      - name: 3.29.2
+        symlink: cmake-v3.29
+  cmake-experimental:
```
Comment (mattgodbolt) [inline bin/yaml/tools.yaml:23]: please indent this whole section once and name it `experimental` - there's no need to have `cmake` and `cmake-experimental` as two top-level packages: `cmake/experimental` is a fine way to separate versions.

I tested this install and it seems to work - so thank you.

File: bin/yaml/tools.yaml [hunk 4]
```diff
@@ -9,13 +9,28 @@ tools:
...
+    type: tarballs
+    compression: gz
+    url: https://cmake.org/files/dev/cmake-{name}-linux-x86_64.tar.gz
+    strip_components: 1
+    dir: cmake-v{name}
+    create_untar_dir: true
+    check_exe: bin/cmake --help
+    targets:
+      - name: 3.29.20240506-g1ea8fa8
+        symlink: cmake-importstd
```
Comment (mattgodbolt) [inline bin/yaml/tools.yaml:33]: please remove this too (this can be just:
```yaml
targets:
  - 3.29.20240506-g1ea8fa8
```

**Review by mattgodbolt — CHANGES_REQUESTED** [review]
Thanks! I'm sorry we don't have good documentation here, but the `symlink` thing is purely a workaround for the site's one "cmake" used for building programs. Can you please remove the other symlinks? And the `ninja` here seems unrelated and unused - if it's necessary can you link it somewhere or explain why it's needed? thanks!

File: bin/yaml/tools.yaml (hunk 2 above)
Comment (mathstuf) [reply bin/yaml/tools.yaml:45]: Oh, I was matching Windows' deployment; I can add it to the other repo as well.

Comment (mattgodbolt) [reply bin/yaml/tools.yaml:45]: What uses this `ninja` version, out of interest?

Comment (mathstuf) [reply bin/yaml/tools.yaml:45]: I just removed it for now.

**Review by mattgodbolt — APPROVED** [review]
Thank you!

**Pull request merged**